ui/app/AppLayouts/Wallet/adaptors/TokenSelectorViewAdaptor.qml

```diff
@@ -0,0 +1,155 @@
+import QtQuick 2.15
+
+import StatusQ 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Utils 0.1
+
+import shared.stores 1.0
+
+import SortFilterProxyModel 0.2
+
+QObject {
+    id: root
+
+    /**
+      Transforms and prepares input data (assets) for TokenSelectorView needs. The assets model is internally
+      joined with `flatNetworksModel` for the `balances` submodel
+
+      Expected assets model structure:
+      - tokensKey: string -> unique string ID of the token (asset); e.g. "ETH" or contract address
+      - name: string -> user visible token name (e.g. "Ethereum")
+      - symbol: string -> user visible token symbol (e.g. "ETH")
+      - decimals: int -> number of decimal places
+      - communityId: string -> optional; ID of the community this token belongs to, if any
+      - marketDetails: var -> object containing props like `currencyPrice` for the computed values below
+      - balances: submodel -> [ chainId:int, account:string, balance:BigIntString, iconUrl:string ]
+
+      Computed values:
+      - currentCurrencyBalance: double (e.g. `1000.42` in user's fiat currency)
+      - currentBalance: double (e.g. `1.42`)
+    */
+
+    // input API
+    required property var assetsModel
+    required property var flatNetworksModel
+    required property CurrenciesStore currencyStore
```

Comment:
I strongly recommend delaying passing the whole stores as long it's reasonable. This store has tens of methods and props so reading API of `TokenSelectorViewAdaptor` now doesn't really says what is needed by this component. It highly violates `ISP` of `SOLID`. Pushing dependency on stores up in the hierarchy will help us with better "layering" of the app and simpler testing.

Reply:
Yeah, not a big fan of passing the whole store around; I'll narrow it down to the only 2 methods and 1 property that we actually need from there

Reply:
Removed the store from the adaptor